src/main/kotlin/org/usecase/app/Router.kt

```diff
@@ -0,0 +1,26 @@
+package org.usecase.app
+
+import io.swagger.v3.oas.annotations.Operation
+import org.usecase.param.DefaultParam
+import org.usecase.user.context.PublicPipe
+import org.usecase.user.process.HealthCheckProcess
+import org.usecase.wrapper.RouterWrapper
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+
+@Path("/healthcheck")
+class Router : RouterWrapper() {
+
+    @GET
+    @Operation(tags = ["HealthCheck"], summary = "Checks status of DB connection and coroutines")
+    fun check() {
+        PublicPipe.apply {
+            handle(connectionPipe, DefaultParam()) { context ->
+                HealthCheckProcess(context).check()
+            }
+            handle(transactionPipe, DefaultParam()) { context ->
```

Comment:
why do we need this twice?

Reply:
One for each pipe. Usecae uses the same datasource for different pipes, but often we use different data sources for writing and reading (read replica). I think it's better if it's already separate, that way just changing the datasource of the pipe will suffice